### PR TITLE
Added aat yaml file changes for fis-ds-update-web application

### PR DESCRIPTION
**Pull request by yogendraU**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ISDB-658

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by jordankainos — DISMISSED** [review]
lgtm

**Review by hmcts-shajahan — APPROVED** [review]

**Review by parvezhmcts — APPROVED** [review]
ok

**Review by hmcts-shajahan — APPROVED** [review]

**Review by parvezhmcts — APPROVED** [review]
ok

File: apps/fis/fis-ds-update-web/aat.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+kind: HelmRelease
+metadata:
+  name: fis-ds-update-web
+  namespace: fis
+spec:
+  values:
+    nodejs:
+      ingressSessionAffinity:
+        enabled: true
+        sessionCookieName: sticky
```
Comment (timja) [inline apps/fis/fis-ds-update-web/aat.yaml:11]: what do you need this for?

**Review by timja — CHANGES_REQUESTED** [review]
do you actually need all these overrides like readiness delay and replicas?

**Review by hmcts-shajahan — APPROVED** [review]

**Review by kunlebalogun — APPROVED** [review]

File: apps/fis/fis-cos-api/aat.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  values:
+    java:
+      replicas: 2
+      memoryRequests: "768Mi"
+      cpuRequests: "1000m"
+      memoryLimits: "3072Mi"
+      cpuLimits: "2500m"
+      readinessDelay: 45
+      readinessTimeout: 5
+      readinessPeriod: 10
```
Comment (timja) [inline apps/fis/fis-cos-api/aat.yaml:15]: these should generally not be set per environment, is there a reason it's set?

File: apps/fis/fis-ds-update-web/demo.yaml [hunk 3]
```diff
@@ -2,10 +2,16 @@ apiVersion: helm.toolkit.fluxcd.io/v2beta1
...
-      ingressClass: traefik-private
-      image: hmctspublic.azurecr.io/fis/ds-update-web:pr-89-d75dc14-20230317141052 #{"$imagepolicy": "flux-system:demo-fis-ds-update-web"}
-      environment:
-        DUMMY_VAR_TO_REDEPLOY: 2
+    nodejs:
+      ingressClass: traefik-no-proxy
+      disableTraefikTls: false
+      image: hmctspublic.azurecr.io/fis/ds-update-web:pr-89-01c0859-20230315164613 #{"$imagepolicy": "flux-system:demo-fis-ds-update-web"}
+      cpuLimits: "1000m"
+      cpuRequests: "500m"
```
[the remark this answers is not in the document]
Comment (yogendraU) [reply apps/fis/fis-ds-update-web/demo.yaml:13]: Hi @timja , as of now, i will remove these attributes. incase after deploying to demo/aat, if i see issues, i will add them

**Comment by timja** [comment]
> > do you actually need all these overrides like readiness delay and replicas?
> 
> We have a feature to upload 50Mb plus files to the case. due to which we need 2 replicas and this is a generic application which will be used by multiple teams to upload documents against a Case

2 replicas should be the default in the base though you shouldn't need to override it

File: apps/fis/fis-cos-api/aat.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  values:
+    java:
+      useInterpodAntiAffinity: true
+      image: hmctspublic.azurecr.io/fis/cos-api:prod-c9e5ffd-20230314190321 #{"$imagepolicy": "flux-system:fis-cos-api"}
+      environment:
+        DUMMY_VAR_TO_REDEPLOY: 2
+      global:
+        environment: aat
+        tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
+        enableKeyVaults: true
```
Comment (timja) [inline apps/fis/fis-cos-api/aat.yaml:15]: not needed
```suggestion
```

File: apps/fis/fis-cos-api/aat.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: fis-cos-api
+spec:
+  values:
+    java:
+      useInterpodAntiAffinity: true
```
Comment (timja) [inline apps/fis/fis-cos-api/aat.yaml:8]: not needed
```suggestion
```

File: apps/fis/fis-cos-api/fis-cos-api.yaml [hunk 6]
```diff
@@ -8,10 +8,6 @@ spec:
   values:
     java:
       image: hmctspublic.azurecr.io/fis/cos-api:prod-c9e5ffd-20230314190321 #{"$imagepolicy": "flux-system:fis-cos-api"}
-      memoryRequests: "2048Mi"
-      cpuRequests: "1000m"
-      memoryLimits: "4096Mi"
-      cpuLimits: "2000m"
```
Comment (timja) [inline apps/fis/fis-cos-api/fis-cos-api.yaml:14]: sorry if I confused you but is there a reason you're removing these from the base file?
I only meant for you to remove from per environment overrides.

(unless these are too much which they do look quite high)

File: apps/fis/fis-ds-update-web/aat.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: fis-ds-update-web
+  namespace: fis
+spec:
+  values:
+    nodejs:
+      ingressSessionAffinity:
+        enabled: true
```
Comment (timja) [inline apps/fis/fis-ds-update-web/aat.yaml:10]: why is this needed?

**Review by timja — COMMENTED** [review]
closer

File: apps/fis/fis-cos-api/fis-cos-api.yaml (hunk 6 above)
Comment (yogendraU) [reply apps/fis/fis-cos-api/fis-cos-api.yaml:14]: oh ok. sorry i might have misunderstood you.. yes. we would need them in base file with that figure. as we are allowing upto 10 files to be uploaded before case is submitted. And the UI will be accessible by Citizens/Legal Professionals/Police

File: apps/fis/fis-cos-api/aat.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+      environment:
+        DUMMY_VAR_TO_REDEPLOY: 2
+  chart:
+    spec:
+      chart: ./stable/fis-cos-api
+      sourceRef:
+        kind: GitRepository
+        name: hmcts-charts
+        namespace: flux-system
+      interval: 1m
```
Comment (timja) [inline apps/fis/fis-cos-api/aat.yaml:17]:
```suggestion
```

File: apps/fis/fis-ds-update-web/fis-ds-update-web.yaml [hunk 9]
```diff
@@ -2,16 +2,15 @@ apiVersion: helm.toolkit.fluxcd.io/v2beta1
...
+  namespace: fis
 spec:
   releaseName: fis-ds-update-web
   interval: 1m
   values:
     nodejs:
+      disableTraefikTls: true
+      useInterpodAntiAffinity: true
       image: hmctspublic.azurecr.io/fis/ds-update-web:prod-58dc492-20230316221023 #{"$imagepolicy": "flux-system:fis-ds-update-web"}
-      memoryRequests: "2048Mi"
```
Comment (timja) [inline apps/fis/fis-ds-update-web/fis-ds-update-web.yaml:11]: same here on the files

File: apps/fis/fis-cos-api/fis-cos-api.yaml (hunk 6 above)
Comment (timja) [reply apps/fis/fis-cos-api/fis-cos-api.yaml:14]: that seems like a lot of resources, are you keeping the files in memory the whole time rather than storing them somewhere?

File: apps/fis/fis-hmc-api/fis-hmc-api.yaml [hunk 10]
```diff
@@ -7,7 +7,6 @@ spec:
   values:
     java:
       disableTraefikTls: true
-      replicas: 2
```
Comment (timja) [inline apps/fis/fis-hmc-api/fis-hmc-api.yaml:10]: you likely didn't want to remove this from the base file

File: apps/fis/fis-ds-update-web/aat.yaml (hunk 7 above)
Comment (yogendraU) [reply apps/fis/fis-ds-update-web/aat.yaml]: i have removed it already in latest commit

File: apps/fis/fis-hmc-api/fis-hmc-api.yaml (hunk 10 above)
Comment (yogendraU) [reply apps/fis/fis-hmc-api/fis-hmc-api.yaml]: yes. i have added them back.

**Review by parvezhmcts — APPROVED** [review]
OK
